src/@types/polyfill.ts

```diff
@@ -0,0 +1,36 @@
+/*
+Copyright 2020 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+export function polyfillTouchEvent() {
+    // Firefox doesn't have touch events, so create a fake one we can rely on lying about.
```

Comment:
To be a bit more specific, Firefox does have them in general, but only exposes `TouchEvent` on platforms like mobile and Windows with touch screen.